submit-web/src/components/DocumentUpload/DocumentTableRow.tsx

```diff
@@ -0,0 +1,169 @@
+    "&:hover": {
+      backgroundColor: BCDesignTokens.themeBlue40,
+    },
+  })
+);
+
+export const StyledTableCell = styled(TableCell)(() => ({
+  borderTop: `1px solid ${BCDesignTokens.themeBlue20}`,
+  borderBottom: `1px solid ${BCDesignTokens.themeBlue20}`,
+  padding: `${BCDesignTokens.layoutPaddingXsmall} !important`,
+  "&:first-of-type": {
+    borderLeft: `1px solid ${BCDesignTokens.themeBlue20}`,
+    borderTopLeftRadius: 5,
+    borderBottomLeftRadius: 5,
+  },
+  "&:last-of-type": {
+    borderRight: `1px solid ${BCDesignTokens.themeBlue20}`,
+    borderTopRightRadius: 5,
+    borderBottomRightRadius: 5,
+  },
+}));
+
+const StyledTableRow = styled(TableRow)(() => ({}));
+
+type StyledTableRowProps = TableRowProps & { error?: boolean };
+export const PackageTableRow = ({
+  error,
+  children,
+  ...otherProps
+}: StyledTableRowProps) => {
+  // pass error to every child
+  const childrenWithProps = React.Children.map(children, (child) =>
+    React.isValidElement(child)
+      ? React.cloneElement(child, { error } as any) // eslint-disable-line @typescript-eslint/no-explicit-any
+      : child
+  );
+
+  return (
+    <StyledTableRow error={error} {...otherProps}>
+      {childrenWithProps}
+    </StyledTableRow>
+  );
+};
+
+type DocumentTableRowProps = {
+  documentItem: {
+    id: number;
+    name: string;
+    submitted_by: string;
+    version: number;
+    url: string;
+  };
+  error?: boolean;
+};
+export default function DocumentTableRow({
+  documentItem,
+  error = false,
+}: DocumentTableRowProps) {
+  const { name, submitted_by, version, url } = documentItem;
+
+  const onActionClick = () => {};
+  const [pendingGetObject, setPendingGetObject] = useState(false);
+
+  const getObjectFromS3 = async () => {
+    try {
+      if (pendingGetObject) return;
+      setPendingGetObject(true);
+      const response = await downloadObject({
+        filename: name,
+        s3sourceuri: url,
+      });
+      const linkUrl = window.URL.createObjectURL(new Blob([response.data]));
+      const link = document.createElement("a");
+      link.href = linkUrl;
+      link.setAttribute("download", name);
+      document.body.appendChild(link);
+      link.click();
+    } catch (e) {
+      notify.error("Failed to download documentItem");
+    } finally {
+      setPendingGetObject(false);
+    }
+  };
+
+  return (
+    <>
+      <PackageTableRow key={`row-${documentItem.name}`} error={error}>
+        <StyledTableCell colSpan={2}>
+          <Typography
+            variant="body1"
+            color="inherit"
+            sx={{
+              overflow: "clip",
+              textOverflow: "ellipsis",
+              cursor: "pointer",
+              mx: 0.5,
+              textDecoration: "none",
+            }}
+          >
+            <MuiLink onClick={getObjectFromS3} sx={{ textDecoration: "none" }}>
+              {name}
+            </MuiLink>
+          </Typography>
+        </StyledTableCell>
```

Comment:
Is there a better naming more related to the document table than StyledTableCell

Reply:
renamed to documenttablecell